sw/cheri/common/block_tests.hh

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright lowRISC contributors.
+ * Licensed under the Apache License, Version 2.0, see LICENSE for details.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+#include "../common/sonata-devices.hh"
+#include <cheri.hh>
+#include <platform-gpio.hh>
+#include <ds/xoroshiro.h>
+
+using namespace CHERI;
+
+// The full definition of Sonata's GPIO aray.
+struct SonataGpioFull {
+  SonataGpioGeneral general;
+  SonataGpioRaspberryPiHat rpi;
+  SonataGpioArduinoShield arduino;
+  SonataGpioPmod pmod;
+};
```

Comment:
Struct of capabilities would be future proof

Reply:
Changed to this, along with appropriate use of `GPIO_BOUNDS` and `GPIO_RANGE` now so that this is resilient to gaps in the GPIO registers being introduced in the future.